src/EnergyPlus/UnitarySystem.cc

```diff
@@ -12864,25 +12864,29 @@ namespace UnitarySystems {
                     }
                     if (this->m_CoolingSpeedNum == 0) this->m_CoolingSpeedNum = 1;
                     bool const singleMode = (this->m_SingleMode == 1);
+                    PartLoadFrac = 1.0;
                     for (int speedNum = this->m_CoolingSpeedNum; speedNum <= this->m_NumOfSpeedCooling; speedNum++) {
                         this->m_CoolingSpeedNum = speedNum;
                         state.dataCoilCooingDX->coilCoolingDXs[this->m_CoolingCoilIndex].simulate(
                             state, OperationMode, PartLoadFrac, this->m_CoolingSpeedNum, this->m_CoolingSpeedRatio, this->m_FanOpMode, singleMode);
                         if ((state.dataLoopNodes->Node(OutletNode).Temp - DesOutTemp) < Acc) break;
                     }
 
-                    std::array<Real64, 8> Par2 = {double(this->m_CoolingCoilIndex),
-                                                  DesOutTemp,
-                                                  // dehumidification mode = 0 for normal mode, 1+ for enhanced mode
-                                                  // need to test what happens when Alt mode doesn't exist, or somehow test for it,
-                                                  // or fatal out in GetInput
-                                                  1.0, // DehumidMode
-                                                  double(FanOpMode),
-                                                  double(this->m_CoolingSpeedNum),
-                                                  1.0, //  this->m_CoolingSpeedRatio;
-                                                  0.0,
-                                                  0.0};
-                    General::SolveRoot(state, Acc, MaxIte, SolFla, PartLoadFrac, &this->genericDXCoilResidual, 0.0, 1.0, Par2);
+                    // make sure outlet temp is below set point before calling SolveRoot
+                    if ((state.dataLoopNodes->Node(OutletNode).Temp - DesOutTemp) < -Acc) {
```

Comment:
The only question I have is whether to negate `Real64 constexpr Acc(1.e-3)` or to add a new `Real64 constexpr minusAcc(-1.e-3)` ?

Reply:
I would personally not add another variable just for that.  In fact, I'd just reverse the subtraction of the two terms so that you don't have to have a negative comparison.

Reply:
I did that math, Outlet - Setpoint < -Acc is same as Setpoint - Outlet < Acc, but when I plugged in numbers I did't get what I expected. Setpoint = 20. As outlet goes lower than 19.999 I want to iterate, not above that temp.

20 - 20.001 = -0.001, which is less than Acc = 0.001 but I don't want to iterate here,
I stopped there and didn't think about it more cause I didn't want to mess up the logic.
20 - 20 = 0
20 - 19.999 = 0.001, it's here I want to start iterating.
20 - 19.998 = 0.002, and also here

It seems I also need to flip the operator when multiplying by -1.

Setpoint - Outlet **>** Acc

I can't recall the 6th grade rules on < and > when switching signs around (i.e., multiplying both sides by -1) but from this data that operator definitely needs to change. And this operator issue vaguely rings a rusty bell now that I walk through this.

So if we agree it's reverse the subtraction AND reverse the operator, I'll go ahead and do that just to etch in stone this logic.

Reply:
You are correct that when you change the sign you have to change the operator, so this sounds good.